### ISSUE TITLE
[Request]: Document alert data for observability rule types

### ISSUE THREAD

**Issue by maryam-saeidi** [elastic/observability-docs#4568]
### 🍒 Summary

We save different alert information for different observability rule types. This information can be used in multiple features such as [maintenance window](https://www.elastic.co/guide/en/kibana/current/maintenance-windows.html) and conditional actions:

<img src="https://github.com/user-attachments/assets/be7f9324-6364-44d9-a1c2-1d96bebc983b" width=500 />

We would like to document the alert data available in all the observability rules and the differences between rules so that our users can get an idea of how to filter their alerts for the above-mentioned features.

### Resources

Here is a [list of fields ](https://docs.google.com/spreadsheets/d/1tVPzZn6RTDfdyHrlsTBSyZbkm_H6szWLG7KI8n-2QHo/edit?usp=sharing)that I am preparing based on fields that we define in code such as [default_alert_as_data](https://github.com/elastic/kibana/blob/main/packages/kbn-rule-data-utils/src/default_alerts_as_data.ts).

### Which documentation set does this change impact?

Stateful and Serverless

### Feature differences

No difference

### What release is this request related to?

N/A

### Collaboration model

The documentation team

### Point of contact.

**Main contact:** @maryam-saeidi 

**Stakeholders:**  @jasonrhodes, @drewpost

**Comment by jasonrhodes** [comment]
I assume we should be engaging @dedemorton for (part of) this?

**Comment by maryam-saeidi** [comment]
Yes! I had a chat with @dedemorton about this topic, and she proposed some places where we can start including this information; we just need to first prepare a list of fields and their description, possibly with an example, that we want to include in the documentation.

**Comment by lcawl** [comment]
It's still needs some pondering but we're proposing to have reference content derivable from yaml files in the new docs system (so that it doesn't require fussy maintenance of tables and lends itself it automation so they get less out-dated). We're playing with this for settings references, for example, in https://github.com/elastic/kibana/pull/191787.  A similar effort might make sense in the case of these field references for each rule.  At a basic level it would require consensus on what properties we'd want in the yaml file and then we'd just work with the docs engineers to ensure it can be output well in the new system.

**Comment by maryam-saeidi** [comment]
Thanks, @lcawl, for sharing the example PR. Having a yaml file for each rule sounds good to me! Regarding coming up with a sample template and list of fields, maybe we can start with a small set of fields, such as name, description, and an example and then extend it in future if needed. Or did you have something else in mind?

**Comment by heespi** [comment]
Adding @mikecote for visibility and inputs here.
